src/services/userService.js

```diff
@@ -0,0 +1,44 @@
+const { data } = require('../data');
+
+const getAll = () => {
+  return data.users;
+};
+
+const getById = (id) => {
+  return data.users.find((user) => user.id === id) || null;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getById(id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
+
+const createUser = (name) => {
+  const user = {
+    name,
+    id: data.nextId++,
+  };
+
+  data.users.push(user);
+
+  return user;
+};
+
+const deleteUser = (id) => {
+  const initialLength = data.users.length;
+
+  data.users = data.users.filter((user) => user.id !== id);
+
+  return data.users.length < initialLength;
+};
```

Comment:
In the deleteUser function, you are modifying the users array by filtering out the user with the specified id. This is a potentially expensive operation as it requires iterating through the entire array. Consider using a different data structure, such as a map or an object, where deletion can be done in constant time.